numba/targets/linalg.py

```diff
@@ -2425,3 +2425,71 @@ def _1d_matrix_rank_impl(a, tol):
             assert 0  # unreachable
 
     return _get_matrix_rank_impl(a, tol)
+
+
+@overload(np.linalg.matrix_power)
+def matrix_power_impl(a, n):
+    """
+    Computes matrix power. Only integer powers are supported in numpy.
+    """
+
+    _check_linalg_matrix(a, "matrix_power")
+    np_dtype = np_support.as_dtype(a.dtype)
+
+    nt = getattr(n, 'dtype', n)
+    if not isinstance(nt, types.Integer):
+        raise TypeError("Exponent must be an integer.")
+
+    def matrix_power_impl(a, n):
+
+        if n == 0:
+            # this should be eye() but it doesn't support
```

Comment:
We should consider exposing an internal version of eye() that supports the dtype arg